src/query.h

```diff
@@ -368,6 +368,14 @@ struct QueryDatabase {
                            size_t symbol_index,
                            const std::string& short_name,
                            const std::string& detailed_name);
+
+  // Query the indexing structure to look up symbol id for given Usr.
+  // If |no_insert| is true, in case the given Usr does not exist
+  // in the indexing structure, |INVALID_ID| will be returned
+  QueryFileId GetQueryFileIdFromPath(const std::string& path, bool no_insert);
```

Comment:
What about returning optional<> instead of INVALID_ID? And I think only querydb code wants to set |no_insert| to false, so that doesn't need to be part of the public interface.

btw, please try to avoid double negatives, so no_insert -> create_if_missing

Reply:
@jacobdufault I made changes to address the points you made. Please take a look at https://github.com/khng300/cquery/commit/2552209d9627211f3882ef44676848031c01ebae